cpp/src/replace/replace.cu

```diff
@@ -81,16 +95,39 @@ namespace{ //anonymous
                     gdf_size_type      nrows,
                     const void* d_old_values,
                     const void* d_new_values,
-                    gdf_size_type      nvalues)
+                    gdf_size_type     nvalues,
+                    gdf_valid_type* col_valid,
+                    gdf_valid_type* new_valid)
     {
+      const bool is_col_valid = (col_valid != nullptr);
+      const bool is_new_valid = (new_valid != nullptr);
+
+      bit_mask::bit_mask_t *typed_col_valid = reinterpret_cast<bit_mask::bit_mask_t*>(col_valid);
+      const bit_mask::bit_mask_t *typed_new_valid = reinterpret_cast<const bit_mask::bit_mask_t*>(new_valid);
+
       thrust::device_ptr<const col_type> old_values_begin = thrust::device_pointer_cast(static_cast<const col_type*>(d_old_values));
 
       const size_t grid_size = nrows / BLOCK_SIZE + (nrows % BLOCK_SIZE != 0);
```

Comment:
Use `cudf::util::cuda::grid_config_1d grid{nrows, BLOCK_SIZE, 1};` from utilities/cuda_utils.hpp  (protects against overflow).

Reply:
changed to `cudf::util::cuda::grid_config_1d`